client/src/components/CatsList/CatList.tsx

```diff
@@ -0,0 +1,85 @@
+import { Fragment, useEffect, useCallback, useState, useRef } from 'react';
+import { useCats } from '../../contexts/cats.context';
+import { useStyles } from './style';
+import { Cat } from './components/Cat';
+import { FilterInput } from '../common/components/FilterInput';
+import { NavLink } from 'react-router-dom';
+import { Button } from '../common/components/Button';
+import type { TCat } from '../../types/cat';
+import debounce from 'lodash.debounce';
+import axios from 'axios';
+
+const PLACE_HOLDER_TEXT: string = 'Search for a cat or mouse';
+const NO_CATS_FOUND: string = 'No Cats Found';
+const ADD_CAT_TEXT: string = 'Add Cat';
+
+export const CatList = () => {
+  const store = useCats();
+  const styles = useStyles();
+  const [filterData, setFilterData] = useState<TCat[]>([]);
+  const [loading, setLoading] = useState<Boolean>(true);
+  const filterRef = useRef<string>('');
+  const abortController = useRef<AbortController>(new AbortController());
```

Comment:
No need for type if I can tell what it is
```suggestion
  const [loading, setLoading] = useState(true);
  const filterRef = useRef('');
  const abortController = useRef(new AbortController());
```